### PR TITLE
New functionalities of Storage.Pickers: SettingsIdentifier; InitialFileTypeIndex; Title; PickMultipleFoldersAsync; ShowOverwritePrompt;

### PR DESCRIPTION
**Pull request by DinahK-2SO**
Implementation for
- #6051

This pull request updates the Storage.Pickers API implementation to add new customization and usability features to the `FileOpenPicker`, `FileSavePicker`, and `FolderPicker` classes. The changes introduce properties for dialog title and instance-specific settings, allow specifying the default file type filter by 0-based index, and add support for picking multiple folders. These enhancements are reflected in both the API definitions and usage examples.

### Title

* Adds `Title` properties to `FileOpenPicker`, `FileSavePicker`, and `FolderPicker`, enabling custom dialog titles. 

Related:
- #5879

### SettingsIdentifier
* Adds `SettingsIdentifier` properties to `FileOpenPicker`, `FileSavePicker`, and `FolderPicker`, enabling instance-specific picker state across sessions. 

Related:
- #5847
- https://github.com/microsoft/microsoft-ui-xaml/issues/10904

### InitialFileTypeIndex

* Adds `InitialFileTypeIndex` property to `FileOpenPicker` and `FileSavePicker`, allowing developers to set the default selected file type filter by index (0-based).

Related:
- #5975

### Changes in FileSavePicker
* Adds a new properties for `FileSavePicker`: `ShowOverwritePrompt`:
  `ShowOverwritePrompt` default to `true` and control whether the picker warns about overwriting when user picked an existing file via FileSavePicker.

* Changing the default behavior - when the user picked a not-existing file via FileSavePicker, the picker will NOT create an empty file there - allowing developer to decide when to make this file.

Related:
- #5976

### FolderPicker.PickMultipleFoldersAsync

* Adds `PickMultipleFoldersAsync` method to `FolderPicker`, enabling selection of multiple folders in a single operation. API definitions, documentation, and code samples have been updated to reflect this feature.

Related:
- #5848

### CODE REVIEW

File: dev/Interop/StoragePickers/PickerCommon.cpp [hunk 1]
```diff
@@ -430,11 +468,26 @@ namespace PickerCommon {
         {
             check_hresult(dialog->SetFileTypes(static_cast<UINT>(FileTypeFilterPara.size()), FileTypeFilterPara.data()));
 
-            if (FocusLastFilter)
+            if (InitialFileTypeIndex != -1 && InitialFileTypeIndex < static_cast<int>(FileTypeFilterPara.size()))
+            {
+                check_hresult(dialog->SetFileTypeIndex(InitialFileTypeIndex + 1)); // COMDLG file type index is 1-based
+            }
```
Comment (yeelam-gordon) [inline dev/Interop/StoragePickers/PickerCommon.cpp:494]: InitialFileTypeIndex is only checked for != -1 and < size before calling SetFileTypeIndex. Negative values other than -1 (e.g., -5) satisfy the condition and produce a large unsigned index, leading to E_INVALIDARG or selecting an unintended filter. Guard for value >= 0 (or validate in the setter to reject values < -1) before applying the index.

File: dev/Interop/StoragePickers/PickerCommon.h [hunk 2]
```diff
@@ -51,6 +53,11 @@ namespace PickerCommon {
         void ConfigureDialog(winrt::com_ptr<IFileDialog> dialog);
         void ConfigureFileSaveDialog(winrt::com_ptr<IFileSaveDialog> dialog);
 
+        winrt::hstring TryGetAppUserModelId();
+        winrt::hstring TryGetProcessFullPath();
+
+		bool ShowOverwritePrompt;
```
Comment (yeelam-gordon) [inline dev/Interop/StoragePickers/PickerCommon.h]: PickerParameters leaves ShowOverwritePrompt uninitialized. Callers that use a default PickerParameters (e.g., ConfigureFileSaveDialog in shared test helpers) may randomly disable overwrite prompting. Initialize the field to true in the struct or set it before use to preserve default prompting behavior.

**Review by yeelam-gordon — COMMENTED** [review]
Functionality review comments for PR 6052.

**Review by yeelam-gordon — APPROVED** [review]

File: dev/Interop/StoragePickers/FileOpenPicker.cpp
Comment (Scottj1s) [inline dev/Interop/StoragePickers/FileOpenPicker.cpp]: don't forget to remove velocity checks when an API is made stable.

File: dev/Interop/StoragePickers/FileOpenPicker.cpp [hunk 3]
```diff
@@ -59,6 +81,16 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
...
     }
+    int FileOpenPicker::InitialFileTypeIndex()
+    {
+        THROW_HR_IF(E_NOTIMPL, !::Microsoft::Windows::Storage::Pickers::Feature_StoragePickers2::IsEnabled());
+        return m_initialFileTypeIndex;
+    }
+    void FileOpenPicker::InitialFileTypeIndex(int value)
+    {
+        THROW_HR_IF(E_NOTIMPL, !::Microsoft::Windows::Storage::Pickers::Feature_StoragePickers2::IsEnabled());
+        m_initialFileTypeIndex = value;
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/FileOpenPicker.cpp:85]: we had discussed in API review validating that this value is [-1,max(count(FileTypeChoices),count(FileTypeFilter)) )

we should at least throw on < -1

File: dev/Interop/StoragePickers/FileOpenPicker.h [hunk 4]
```diff
@@ -52,6 +63,8 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
             }()
         };
 
+        int m_initialFileTypeIndex{ -1 };
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/FileOpenPicker.h]:
```suggestion
        constexpr int DefaultInitialFileType{ -1 };
        ...
        int m_initialFileTypeIndex{ DefaultInitialFileType };
```

File: dev/Interop/StoragePickers/FileSavePicker.cpp
Comment (Scottj1s) [inline dev/Interop/StoragePickers/FileSavePicker.cpp]: same: can remove all Feature_StoragePickers refs now

File: dev/Interop/StoragePickers/FolderPicker.cpp [hunk 5]
```diff
@@ -131,4 +153,75 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
         logTelemetry.Stop(m_telemetryHelper, true);
         co_return result;
     }
+
+    winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Foundation::Collections::IVectorView<winrt::Microsoft::Windows::Storage::Pickers::PickFolderResult>> FolderPicker::PickMultipleFoldersAsync()
+    {
+        // TODO: remove get strong reference when telementry is safe stop
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/FolderPicker.cpp]: what is this TODO? can we remove either the code or the comment?

File: dev/Interop/StoragePickers/FolderPicker.cpp [hunk 6]
```diff
@@ -131,4 +153,75 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
         logTelemetry.Stop(m_telemetryHelper, true);
         co_return result;
     }
+
+    winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Foundation::Collections::IVectorView<winrt::Microsoft::Windows::Storage::Pickers::PickFolderResult>> FolderPicker::PickMultipleFoldersAsync()
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/FolderPicker.cpp:145]: I don't see a test for PickMultipleFoldersAsync

File: dev/Interop/StoragePickers/FolderPicker.cpp [hunk 7]
```diff
@@ -131,4 +153,75 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
...
+    {
+        // TODO: remove get strong reference when telementry is safe stop
+        auto lifetime{ get_strong() };
+
+        auto logTelemetry{ StoragePickersTelemetry::FolderPickerPickMultipleFolder::Start(m_telemetryHelper) };
+
+        // capture parameters to avoid using get strong referece of picker
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/FolderPicker.cpp]: why? caller still needs to hold the picker to retrieve results

File: dev/Interop/StoragePickers/FolderPicker.cpp [hunk 8]
```diff
@@ -131,4 +153,75 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
...
+            co_return results.GetView();
+        }
+
+        auto dialog = create_instance<IFileOpenDialog>(CLSID_FileOpenDialog, CLSCTX_INPROC_SERVER);
+
+        FILEOPENDIALOGOPTIONS dialogOptions;
+        check_hresult(dialog->GetOptions(&dialogOptions));
+        check_hresult(dialog->SetOptions(dialogOptions | FOS_PICKFOLDERS | FOS_FORCEFILESYSTEM | FOS_ALLOWMULTISELECT));
+
+        {
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/FolderPicker.cpp:174]: just to scope hr? probably not needed

File: dev/Interop/StoragePickers/FolderPicker.cpp [hunk 9]
```diff
@@ -131,4 +153,75 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
...
+            auto hr = dialog->Show(parameters.HWnd);
+            if (FAILED(hr) || cancellationToken())
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/FolderPicker.cpp]: or just
```suggestion
            if (FAILED(dialog->Show(parameters.HWnd)) || cancellationToken())
```

File: dev/Interop/StoragePickers/FolderPicker.cpp [hunk 10]
```diff
@@ -131,4 +153,75 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
...
+
+        if (results.Size() > 0)
+        {
+            logTelemetry.Stop(m_telemetryHelper, isCancelled, true);
+        }
+        else
+        {
+            logTelemetry.Stop(m_telemetryHelper, isCancelled, false);
+        }
+        co_return resultView;
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/FolderPicker.cpp]: why the local?
```suggestion
        co_return results.GetView();
```

File: dev/Interop/StoragePickers/FileOpenPicker.cpp [hunk 11]
```diff
@@ -86,9 +118,14 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
...
         parameters.CommitButtonText = m_commitButtonText;
+        parameters.Title = m_title;
+        parameters.SettingsIdentifier = m_settingsIdentifier;
         parameters.SuggestedFolder = m_suggestedFolder;
         parameters.SuggestedStartLocation = m_suggestedStartLocation;
         parameters.SuggestedStartFolder = m_suggestedStartFolder;
+        
+        // Note: write the user-defined index before capturing filter spec so the filter logic can honor it.
+        parameters.InitialFileTypeIndex = m_initialFileTypeIndex;
         parameters.CaptureFilterSpecData(m_fileTypeFilter.GetView(), m_fileTypeChoices.GetView());
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/FileOpenPicker.cpp]: this could be a good opportunity to validate that m_initialFileTypeIndex < max(count(FilteTypeFilter),count(FileTypeChoices))

but we should definitely do it somewhere before the dialog is presented

File: dev/Interop/StoragePickers/PickerCommon.cpp [hunk 12]
```diff
@@ -356,7 +357,11 @@ namespace PickerCommon {
             FileTypeFilterPara.push_back({ FileTypeFilterData.at(i * 2).c_str(), FileTypeFilterData.at(i * 2 + 1).c_str() });
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/PickerCommon.cpp]: nit: "para" suggests "paragraph", where as "param(s)" suggests parameter(s)

File: dev/Interop/StoragePickers/PickerCommon.cpp [hunk 13]
```diff
@@ -392,13 +397,46 @@ namespace PickerCommon {
...
+        wchar_t appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+        UINT32 appUserModelIdSize{ APPLICATION_USER_MODEL_ID_MAX_LENGTH };
+
+        auto hr{GetCurrentApplicationUserModelId(&appUserModelIdSize, appUserModelId) };
+        if (SUCCEEDED(hr))
+        {
+            return winrt::hstring{ appUserModelId };
+        }
+
+        return winrt::hstring{};
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/PickerCommon.cpp]:
```suggestion
        return {};
```

File: dev/Interop/StoragePickers/PickerCommon.cpp [hunk 14]
```diff
@@ -392,13 +397,46 @@ namespace PickerCommon {
...
+    }
+
+    winrt::hstring PickerParameters::TryGetProcessFullPath()
+    {
+        PCWSTR exeFullPath{};
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/PickerCommon.cpp]: don't need local

File: dev/Interop/StoragePickers/PickerCommon.cpp [hunk 15]
```diff
@@ -392,13 +397,46 @@ namespace PickerCommon {
         }
     }
 
+    winrt::hstring PickerParameters::TryGetAppUserModelId()
+    {
+        wchar_t appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH] = {};
+        UINT32 appUserModelIdSize{ APPLICATION_USER_MODEL_ID_MAX_LENGTH };
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/PickerCommon.cpp]: safer (single source of truth):
```suggestion
        UINT32 appUserModelIdSize{ _countof(appUserModelId) };
```

File: dev/Interop/StoragePickers/PickerCommon.cpp [hunk 16]
```diff
@@ -430,11 +468,26 @@ namespace PickerCommon {
...
+        }
+
+        if (!IsHStringNullOrEmpty(SettingsIdentifier))
+        {
+            auto appDistinctString = TryGetAppUserModelId();
+            if (appDistinctString.empty())
+            {
+                appDistinctString = TryGetProcessFullPath();
+            }
+            if (!appDistinctString.empty())
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/PickerCommon.cpp:504]: if both TryGetAppUserModelId and TryGetProcessFullPath failed (unlikely, I admit), then SettingsIdentifier fails silently.  I'd throw instead.

File: test/StoragePickersTests/PickerCommonTests.cpp [hunk 17]
```diff
@@ -250,6 +250,53 @@ namespace Test::PickerCommonTests
                 L"*.txt;*.doc;*.docx");
         }
 
+        TEST_METHOD(VerifyFileOpenPickerInitialFileTypeIndexApplied)
```
Comment (Scottj1s) [inline test/StoragePickersTests/PickerCommonTests.cpp]: need tests to verify failure results too e.g., InitialFileTypeIndex out of range, but other params too.

File: test/StoragePickersTests/PickerCommonTests.cpp [hunk 18]
```diff
@@ -283,6 +330,35 @@ namespace Test::PickerCommonTests
                 L"*.txt;*.doc;*.docx");
         }
 
+        TEST_METHOD(VerifyFileSavePickerInitialFileTypeIndexApplied)
```
Comment (Scottj1s) [inline test/StoragePickersTests/PickerCommonTests.cpp]: just testing FileTypeFilter with FileOpen and FilteTypeChoices with FileSave - sufficient?

File: test/StoragePickersTests/PickerCommonTests.cpp [hunk 19]
```diff
@@ -535,6 +611,31 @@ namespace Test::PickerCommonTests
...
+        {
+            // Arrange: start with prompt enabled (default behavior).
+            PickerParameters promptOnParameters{};
+            promptOnParameters.ShowOverwritePrompt = true;
+            auto dialogWithPrompt = winrt::create_instance<IFileSaveDialog>(CLSID_FileSaveDialog, CLSCTX_INPROC_SERVER);
+            promptOnParameters.ConfigureDialog(dialogWithPrompt.as<IFileDialog>());
+            promptOnParameters.ConfigureFileSaveDialog(dialogWithPrompt);
+
+            FILEOPENDIALOGOPTIONS promptOptions{};
+            dialogWithPrompt->GetOptions(&promptOptions);
```
Comment (Scottj1s) [inline test/StoragePickersTests/PickerCommonTests.cpp]: test code should VERIFY all hresults

File: dev/Interop/StoragePickers/FileOpenPicker.cpp [hunk 20]
```diff
@@ -86,9 +118,14 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
     {
         parameters.HWnd = winrt::Microsoft::UI::GetWindowFromWindowId(m_windowId);
         parameters.CommitButtonText = m_commitButtonText;
+        parameters.Title = m_title;
+        parameters.SettingsIdentifier = m_settingsIdentifier;
         parameters.SuggestedFolder = m_suggestedFolder;
         parameters.SuggestedStartLocation = m_suggestedStartLocation;
         parameters.SuggestedStartFolder = m_suggestedStartFolder;
+        
+        // Note: write the user-defined index before capturing filter spec so the filter logic can honor it.
```
Comment (codendone) [inline dev/Interop/StoragePickers/FileOpenPicker.cpp]: > // Note: write the user-defined index before capturing filter spec so the filter logic can honor it.
This feels like a potential tripping hazard. Consider passing m_initialFileTypeIndex into CaptureFilterSpecData.

File: dev/Interop/StoragePickers/FileSavePicker.cpp [hunk 21]
```diff
@@ -171,10 +218,6 @@ namespace winrt::Microsoft::Windows::Storage::Pickers::implementation
         check_hresult(shellItem->GetDisplayName(SIGDN_NORMALDISPLAY, fileName.put()));
         std::wstring fileNameStr(fileName.get());
 
-        // Create an empty file if the file doesn't exist,
-        // If the file already exists, do nothing.
-        auto [handle, _] = wil::try_open_or_create_file(pathStr.c_str(), GENERIC_WRITE);
```
Comment (codendone) [inline dev/Interop/StoragePickers/FileSavePicker.cpp:176]: What's this change? This seems like a behavior change separate from the new features?

Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FileSavePicker.cpp:176]: Thanks @codendone for raising this question! Here we're changing the current behavior of FileSavePicker to be:
- when the user picked a not-existing file via FileSavePicker, the picker will return the result with user-defined file path and will NOT create an empty file there - allowing developer to decide when to make this file.

Below is its background story:

The UWP FileSavePicker creates an empty file when a non-existing file is chosen. This makes sense in the context of UWP, where the picker returns a StorageFile object and expects the backing file to exist.

Our newer pickers return a lightweight object with a string path. In this newer model, creating an empty file up front is no longer necessary.

We followed the behavior of UWP picker in 1.8. However, based on developers' feedback, this automatic creation of an empty file can cause confusion, and, sometimes developers need an option to disable this automatic creation (#5976)

Instead of adding such an option, it would be better to remove the automatic file creation by default - this is a decision during API review, where reviewers consider WinAppSDK 2.0 a good opportunity for us to make this behavior change.

Therefore, we're removing the file creation code here.

File: dev/Interop/StoragePickers/PickerCommon.cpp (hunk 1 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/PickerCommon.cpp:494]: Thanks @yeelam-gordon for catching this! Added validation.

File: dev/Interop/StoragePickers/PickerCommon.h (hunk 2 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/PickerCommon.h]: Thanks for catching this. Updated code with init value!

File: dev/Interop/StoragePickers/FileOpenPicker.cpp (hunk 20 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FileOpenPicker.cpp]: Thanks @codendone for catching this. Updated the method parameters.

**Comment by DinahK-2SO** [comment]
I have kicked off a TouchDown pipeline, will upload the localization files for the validation message of InitialFileTypeIndex when they're ready.

File: dev/Interop/StoragePickers/PickerCommon.cpp (hunk 16 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/PickerCommon.cpp:504]: `SettingsIdentifier` is an optional property. Would failing the entire picker because we couldn’t derive a `ClientGuid` be too strong? 

I am not sure about this change, should we throw error if appid and file path both unavailable or keep the best‑effort behavior instead? 

I am keeping this as an open question and adding @codendone , @yeelam-gordon and @Hong-Xiang for more insights.

File: dev/Interop/StoragePickers/FileOpenPicker.cpp
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FileOpenPicker.cpp]: Thanks @Scottj1s for raising this point! I have updated the files.

File: dev/Interop/StoragePickers/FileOpenPicker.cpp (hunk 3 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FileOpenPicker.cpp:85]: Added validation. on values < -1

File: dev/Interop/StoragePickers/FolderPicker.cpp (hunk 5 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FolderPicker.cpp]: Thanks @Scottj1s for catching this. This is from a legacy comment in existing picker methods. It's "TODO" format is indeed confusing.

I am updating the comment here to be:

> // Keep a strong ref so m_telemetryHelper (a member of the picker) stays valid after co_awaits.


for all picker methods with this comment.

File: dev/Interop/StoragePickers/PickerCommon.cpp (hunk 15 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/PickerCommon.cpp]: Updated!

File: dev/Interop/StoragePickers/FileSavePicker.cpp
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FileSavePicker.cpp]: removed!

File: dev/Interop/StoragePickers/FolderPicker.cpp (hunk 6 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FolderPicker.cpp:145]: Thanks @Scottj1s for raising this. I added a manual test for it, as our test framework doesn't support launching the file dialog.

File: dev/Interop/StoragePickers/FolderPicker.cpp (hunk 8 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FolderPicker.cpp:174]: Thanks! 

Updated this part to be inline for all picker methods.

File: dev/Interop/StoragePickers/FolderPicker.cpp (hunk 9 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FolderPicker.cpp]: Thanks! Updated for all picker methods

File: dev/Interop/StoragePickers/FolderPicker.cpp (hunk 10 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FolderPicker.cpp]: Thanks for catching this! updated for both PickMultipleFilesAsync and PickMultipleFoldersAsync.

File: dev/Interop/StoragePickers/PickerCommon.cpp (hunk 12 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/PickerCommon.cpp]: I see :D 
updated in all files!

File: dev/Interop/StoragePickers/PickerCommon.cpp (hunk 13 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/PickerCommon.cpp]: updated!

File: dev/Interop/StoragePickers/PickerCommon.cpp (hunk 14 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/PickerCommon.cpp]: Thanks! Updated.

File: dev/Interop/StoragePickers/FolderPicker.cpp (hunk 7 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FolderPicker.cpp]: Thanks @Scottj1s for catching this. It is indeed a confusing comment - I removed it.

the purpose of parameters object is to configure dialog.

File: dev/Interop/StoragePickers/FileOpenPicker.cpp (hunk 11 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/FileOpenPicker.cpp]: Thanks @Scottj1s for raising this point! I added the validation inside this method CaptureFilterSpecData.

File: test/StoragePickersTests/PickerCommonTests.cpp (hunk 17 above)
Comment (DinahK-2SO) [reply test/StoragePickersTests/PickerCommonTests.cpp]: Added the test case for index out of range.

File: test/StoragePickersTests/PickerCommonTests.cpp (hunk 19 above)
Comment (DinahK-2SO) [reply test/StoragePickersTests/PickerCommonTests.cpp]: Thanks! updated

File: test/StoragePickersTests/PickerCommonTests.cpp (hunk 18 above)
Comment (DinahK-2SO) [reply test/StoragePickersTests/PickerCommonTests.cpp]: added more test cases for the corner cases!

File: dev/Interop/StoragePickers/ManualTests/README.md [hunk 22]
```diff
@@ -56,5 +56,24 @@ auto& file = co_await picker.PickSaveFileAsync();
     **Example of unexpected behavior - without the fix:**
     ![alter-extension-with-choices-beforefix](./media/alter-extension-with-choices-beforefix.gif)
 
+### FolderPicker.PickMultipleFoldersAsync
+Test code (C++):
+```C++
+winrt::Microsoft::Windows::Storage::Pickers::FolderPicker picker{ AppWindow().Id() };
+auto& results = co_await picker.PickMultipleFoldersAsync();
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/ManualTests/README.md]: should not be auto&, just auto - otherwise, you're taking a ref to a temp object.

File: dev/Interop/StoragePickers/ManualTests/README.md [hunk 23]
```diff
@@ -56,5 +56,24 @@ auto& file = co_await picker.PickSaveFileAsync();
...
 
+// Manually select some folders.
 
+if (results.Size() == 0)
+{
+    std::wcout << L"no folders picked" << std::endl;
+}
+else
+{
+    for (auto const& result : results)
```
Comment (Scottj1s) [inline dev/Interop/StoragePickers/ManualTests/README.md:73]: (this auto & is different, since we're wanting to iterate the lvalue 'results' by reference)
